resolver-dns/src/main/java/io/netty/resolver/dns/DefaultDnsCache.java

```diff
@@ -41,12 +43,53 @@
  */
 @UnstableApi
 public class DefaultDnsCache implements DnsCache {
-
-    private final ConcurrentMap<String, Entries> resolveCache = PlatformDependent.newConcurrentHashMap();
-
     // Two years are supported by all our EventLoop implementations and so safe to use as maximum.
     // See also: https://github.com/netty/netty/commit/b47fb817991b42ec8808c7d26538f3f2464e1fa6
     private static final int MAX_SUPPORTED_TTL_SECS = (int) TimeUnit.DAYS.toSeconds(365 * 2);
+
+    private static final ScheduledFuture<?> CANCELLED = new ScheduledFuture<Object>() {
+
+        @Override
+        public boolean cancel(boolean mayInterruptIfRunning) {
+            return false;
+        }
+
+        @Override
+        public long getDelay(TimeUnit unit) {
+            return Long.MIN_VALUE;
```

Comment:
consider adding a comment saying that we explicitly don't want to follow the interface here and we ignore `unit` so that we return the minimum value for comparison purposes ... the units are always expected to be SECONDS ... etc...